### ISSUE TITLE
Inconsistent/missing CorrectForm and Typo annotations in multi-word tokens

### ISSUE THREAD

**Issue by rhdunn** [UniversalDependencies/UD_English-GUMReddit#5]
The following are inconsistent/missing Typo and CorrectForm annotations. My validation tool is producing confusing output as a result on some of the inconsistent/non-standard annotations:

```
ERROR: Sentence GUM_reddit_bobby-48 token 11 -- unexpected multi-word token 'they're' part form 'None', expected ''re'
ERROR: Sentence GUM_reddit_space-10 token 7 -- unexpected multi-word token 'into's' part upos 'ADP', expected 'NOUN|PROPN|NUM'
ERROR: Sentence GUM_reddit_space-10 token 8 -- unexpected multi-word token 'into's' part form 'None', expected ''s'
```

These appears to be a non-standard use of `CorrectForm=_` to denote the deletion of this word. The https://universaldependencies.org/u/overview/typos.html page under "Extra Word" indicates that such cases should be marked as `reparandum`. See also reviews-018562-0004 in EWT that follows that markup. -- Note also that that does not mark the group as a MWT.

```
ERROR: Sentence GUM_reddit_callout-34 token 1-2 -- unrecognized multi-word token form 'I'veve'
ERROR: Sentence GUM_reddit_callout-35 token 10-11 -- unrecognized multi-word token form 'I'dd'
ERROR: Sentence GUM_reddit_callout-47 token 6-7 -- unrecognized multi-word token form 'it'ss'
ERROR: Sentence GUM_reddit_callout-48 token 19-20 -- unrecognized multi-word token form 'I'mm'
```

These are using e.g. `CorrectForm=I've` on the first token in the MWT to denote the full MWT, and `CorrectForm=_` on the suffix. In the case of typos, the other treebanks have kept the first token/stem as is, and used e.g. `CorrectForm='ve` on the suffix. This matches the guidelines on the typos page.

`ERROR: Sentence GUM_reddit_racial-16 token 1-2 -- unrecognized multi-word base form '"' for suffix 'is'`

This is a wrongly placed MWT. -- It should be on tokens 2-3.

```
ERROR: Sentence GUM_reddit_space-11 token 30-31 -- unrecognized multi-word token form 'its'
```

This is missing a `Typo=Yes` and `CorrectForm='s` on token 31.

**Comment by amir-zeldes** [comment]
> These appears to be a non-standard use of CorrectForm=_ to denote the deletion of this word. The https://universaldependencies.org/u/overview/typos.html page under "Extra Word" indicates that such cases should be marked as reparandum

You are correct that we have been using `CorrectForm=_` to indicate superfluous material - I see what you are saying about `reparandum`, but I don't think that's exactly right here either, since this word isn't being repaired by anything. If this were reparandum, what word would it be attached to? The syntax tree also makes some kind of attempt to do justice to the structure as uttered, so that "are" is actually analyzed as a copula, which reparandum would destroy. The CorrectForm annotation is just trying to express what this sentence might have looked like if it were correct, while the tree expresses the analysis of what is actually there.

> These are using e.g. CorrectForm=I've on the first token in the MWT to denote the full MWT, and CorrectForm=_ on the suffix. In the case of typos, the other treebanks have kept the first token/stem as is, and used e.g. CorrectForm='ve on the suffix

Agreed, that's better, will fix.

> This is a wrongly placed MWT. -- It should be on tokens 2-3.

This is a bug in the GUM build bot, will fix

> This is missing a Typo=Yes and CorrectForm='s on token 31.

Thanks for reporting, will fix

**Comment by rhdunn** [comment]
It would be useful to have the use of `CorrectForm=_` documented on the typo page so its use can be agreed on across treebanks and implemented by tools that e.g. extract/process corrected text.

**Comment by nschneid** [comment]
For written genres without dialogue or other transcribed speech, true speech repair will be virtually nonexistent. I think the typo guidelines are saying `reparandum` should be interpreted more broadly. We have to assign a deprel anyway, so instead of guessing how the word might have attached in a proper syntactic construction, I'd rather just assign `reparandum` to indicate it's a superfluous word.

**Comment by amir-zeldes** [comment]
I'm not sure that's right here - what is the "'re" a reparandum of? And the tree is valid, i.e. the annotator did assign a structure to it; the Typo annotation just tells us that this sentence is not grammatical. Maybe the discrepancy comes from the fact that underlyingly, GUM doesn't have Typo annotations so much as error annotations: anything perceived as incorrect or substantially non-standard English is marked with `<sic>` tags, and we get a target hypothesis in the attribute `ana`. In the UD export, we just shove this into Typo+CorrectForm, but the concept is a little different, and definitely not 1:1 with `reparandum` (which is used in the spoken genres as intended).

**Comment by nschneid** [comment]
> I'm not sure that's right here - what is the "'re" a reparandum of?

Sorry which tree is this referring to? Could you post a screenshot?

**Comment by amir-zeldes** [comment]
![image](https://github.com/UniversalDependencies/UD_English-GUMReddit/assets/2188635/eb3af12a-31e8-4a2c-a9d6-a4f5bf43bc67)

**Comment by nschneid** [comment]
Ah, so this looks like the speaker changed syntactic gears partway through the sentence. There are a couple of nasty sentences like this in EWT that keep causing problems for validation, e.g.:

- I would really love to hear what the French or those who had dated the French to say something about this.

Not sure UD sees its job as correcting the grammar in such cases, but I'm certainly open to expanding the standard if that seems feasible.

**Comment by amir-zeldes** [comment]
Yeah, I agree it's not UD's job to correct the grammar - I think the basic UD tree here is doing a decent job of handling the existing structure. It's just that the `<sic ana="...">` annotation in GUM is based on the concept of target hypotheses, so it gives a 'corrected' rendition of any error or unexpected structure, and the best place to put it in conllu seemed to be Typo/CorrectForm. Maybe this isn't the best way to serialize it, but in most cases it serves the intended purpose of Typo, so it seemed like a decent idea.
